### PR TITLE
Google auth

### PR DESCRIPTION
**Pull request by yujinjcho**
for https://github.com/yujinjcho/rcruitme/issues/13

### CODE REVIEW

**Comment by yujinjcho** [comment]
lets you either create account with email/pw and then sync with google or the other way around

File: app/controllers/SocialAuthController.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import com.mohiva.play.silhouette.impl.providers._
+import javax.inject.Inject
+import play.api.i18n.{ I18nSupport, Messages }
+import play.api.mvc.{ AbstractController, AnyContent, ControllerComponents, Request }
+
+import models.services.UserService
+import utils.auth.DefaultEnv
+
+class SocialAuthController @Inject() (
+    components: ControllerComponents,
```
Comment (achiu8) [inline app/controllers/SocialAuthController.scala]: `cc` for consistency?

File: app/controllers/SocialAuthController.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            authenticator <- silhouette.env.authenticatorService.create(profile.loginInfo)
+            value <- silhouette.env.authenticatorService.init(authenticator)
+            result <- silhouette.env.authenticatorService.embed(value, Redirect(routes.HomeController.google))
+          } yield {
+            silhouette.env.eventBus.publish(LoginEvent(user, request))
+            result
+          }
+        }
+      case _ => Future.failed(new ProviderException(s"Cannot authenticate with unexpected social provider $provider"))
+    }).recover {
```
Comment (achiu8) [inline app/controllers/SocialAuthController.scala]: Does the above block need to be wrapped in parens to chain the `recover`? The one in `SignInController` isn't.

File: app/controllers/SocialAuthController.scala [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+  def authenticate(provider: String) = Action.async { implicit request: Request[AnyContent] =>
+    (socialProviderRegistry.get[SocialProvider](provider) match {
+      case Some(p: SocialProvider with CommonSocialProfileBuilder) =>
+        p.authenticate().flatMap {
+          case Left(result) => Future.successful(result)
+          case Right(authInfo) => for {
+            profile <- p.retrieveProfile(authInfo)
+            user <- userService.save(profile)
+            authInfo <- authInfoRepository.save(profile.loginInfo, authInfo)
```
Comment (achiu8) [inline app/controllers/SocialAuthController.scala]: How are you able to reassign `authInfo` here and is it being used? If not, should we prefix with / use `_` to be explicit?

File: app/models/daos/UserDAO.scala [hunk 4]
```diff
@@ -41,35 +43,73 @@ class UserDAO @Inject()(dbapi: DBApi)(implicit ec: DatabaseExecutionContext) {
...
+            UPDATE users
+              SET
+                first = {first},
+                last = {last},
+                type = {userType},
+                google_key = {googleKey}
+              WHERE
+                id = {userId}
+          """)
+            .on(
```
Comment (achiu8) [inline app/models/daos/UserDAO.scala]: Worth creating `anorm.ToParameterList` instance for `User` so can use `.bind` like with `Job`?

File: app/controllers/SocialAuthController.scala (hunk 1 above)
Comment (yujinjcho) [reply app/controllers/SocialAuthController.scala]: doh

File: app/controllers/SocialAuthController.scala (hunk 3 above)
Comment (yujinjcho) [reply app/controllers/SocialAuthController.scala]: local scope i think. the for loop is converted to flatMap statements. https://docs.scala-lang.org/tutorials/FAQ/yield.html, but ya `_` sounds good

File: app/controllers/SocialAuthController.scala (hunk 2 above)
Comment (yujinjcho) [reply app/controllers/SocialAuthController.scala]: tried removing, but seems like you need it. guess you can't get the result value from the block after match?

**Pull request merged**